homeassistant/components/ecobee/climate.py

```diff
@@ -562,17 +564,18 @@ def extra_state_attributes(self):
     @property
     def is_aux_heat(self):
         """Return true if aux heater."""
-        return "auxHeat" in self.thermostat["equipmentStatus"]
+        return self.thermostat["settings"]["hvacMode"] == ECOBEE_AUX_HEAT_ONLY
```

Comment:
Should this only appear if the device is a heat pump?

See https://www.ecobee.com/home/developer/api/documentation/v1/objects/Thermostat.shtml

```
Note: If no equipment is currently running an empty String is returned. If Settings.hasHeatPump is true, heatPump value will be returned for heating, compCool for cooling, and auxHeat for aux heat. If Settings.hasForcedAir or Settings.hasBoiler is true, auxHeat value will be returned for heating and compCool for cooling (heatPump will not show up for heating).
```

Reply:
the line I removed didn't appear to work as far I can tell (from my theromstat with a heat pump)

the code I added will always return `false` if the HVAC mode is not "auxHeatOnly", so it shouldn't be relevant if the thermostat has a heat pump or not.

You bring up an interesting point that right now Home Assistant will allow any thermostat to attempt to set HVAC mode to "auxHeatOnly".  I assume ecobee API will return an error and keep it in previous heat mode in this case.  I'm not sure if it makes sense to have a check on the Home Assistant side for this though.  Perhaps this could be explored in a future PR?

Reply:
@bdraco Let me know if you see an issue with this approach or have a suggested alternative. If not, could you please approve your review?

Reply:
We should only set the `SUPPORT_AUX_HEAT` supported feature flag if the device actually supports it.

Otherwise I think this is fine.

Reply:
@bdraco ok I made the change you suggested so that the `SUPPORT_AUX_HEAT` flag is only set if the Ecobee has a heat pump.  Please review